### PR TITLE
Add Kameletbinding test to olm_upgrade_tes

### PR DESCRIPTION
**Pull request by lfabriko**

### CODE REVIEW

**Review by squakez — COMMENTED** [review]
Why are you commenting the upgrade test? I think we don't want to remove it.

**Comment by lfabriko** [comment]
Sorry, it was by mistake.

File: e2e/install/olm/olm_upgrade_test.go [hunk 1]
```diff
@@ -196,40 +201,49 @@ func TestOLMAutomaticUpgrade(t *testing.T) {
...
-			Expect(Kamel("delete", "--all", "-n", ns).Execute()).To(Succeed())
-			Expect(Kamel("uninstall", "-n", ns).Execute()).To(Succeed())
-			// Clean up cluster-wide resources that are not removed by OLM
-			Expect(Kamel("uninstall", "--all", "-n", ns, "--olm=false").Execute()).To(Succeed())
+			Eventually(IntegrationConditionStatus(ns, kbindName, v1.IntegrationConditionReady), TestTimeoutLong).
+				Should(Equal(corev1.ConditionTrue))
 		})
+		// Clean up
+		Expect(Kamel("delete", "--all", "-n", ns).Execute()).To(Succeed())
+		Expect(Kamel("uninstall", "-n", ns).Execute()).To(Succeed())
```
Comment (squakez) [inline e2e/install/olm/olm_upgrade_test.go:245]: Uninstall is not required. Better remove it.

File: e2e/install/olm/olm_upgrade_test.go [hunk 2]
```diff
@@ -196,40 +201,49 @@ func TestOLMAutomaticUpgrade(t *testing.T) {
...
+		// Clean up cluster-wide resources that are not removed by OLM
+		Expect(Kamel("uninstall", "--all", "-n", ns, "--olm=false").Execute()).To(Succeed())
```
Comment (squakez) [inline e2e/install/olm/olm_upgrade_test.go:247]: This neither

File: e2e/install/olm/olm_upgrade_test.go [hunk 3]
```diff
@@ -196,40 +201,49 @@ func TestOLMAutomaticUpgrade(t *testing.T) {
...
 			Eventually(IntegrationPodPhase(ns, name), TestTimeoutLong).Should(Equal(corev1.PodRunning))
+			Eventually(IntegrationPodPhase(ns, kbindName), TestTimeoutLong).Should(Equal(corev1.PodRunning))
+
 			Eventually(IntegrationConditionStatus(ns, name, v1.IntegrationConditionReady), TestTimeoutMedium).
 				Should(Equal(corev1.ConditionTrue))
+			Eventually(IntegrationConditionStatus(ns, kbindName, v1.IntegrationConditionReady), TestTimeoutMedium).Should(Equal(corev1.ConditionTrue))
 
 			// Check the Integration version has been upgraded
 			Eventually(IntegrationVersion(ns, name)).Should(ContainSubstring(newIPVersionPrefix))
+			Eventually(IntegrationVersion(ns, kbindName)).Should(ContainSubstring(newIPVersionPrefix))
```
[the remark this answers is not in the document]
Comment (lfabriko) [reply e2e/install/olm/olm_upgrade_test.go:213]: I'm not sure if could be left some checks of underlying integration (also I noticed original checks of integration from L198 and L199 are repeated in the end)...

**Review by squakez — APPROVED** [review]

**Comment by squakez** [comment]
It seems we have some errors to check:
```
      olm_upgrade_test.go:207: 
          Timed out after 900.001s.
          Expected
              <v1alpha1.KameletBindingPhase>: Ready
          to equal
              <v1alpha1.KameletBindingPhase>: Creating
```

**Comment by lfabriko** [comment]
It seems the `Creating` phase is completed too quickly - prior to the check is even run... I tried removing the check...

**Comment by lfabriko** [comment]
May I ask, as branches `main` and `release-1.10.x` diverge in location of `olm_upgrade_test.go` and names of used functions (`C/clusterServiceVersion`), do I understand correctly I need to prepare another MR for branch `release-1.10.x`?

**Comment by squakez** [comment]
> May I ask, as branches `main` and `release-1.10.x` diverge in location of `olm_upgrade_test.go` and names of used functions (`C/clusterServiceVersion`), do I understand correctly I need to prepare another MR for branch `release-1.10.x`?

Yes, very likely.

**Comment by lfabriko** [comment]
MR to `release-1.10.x`: https://github.com/apache/camel-k/pull/4323

**Comment by claudio4j** [comment]
Is there going to be some work on this PR or can we close it ?

**Comment by lfabriko** [comment]
@claudio4j Yes, please don't close it yet, I should rework it due to chages in naming.

**Comment by lfabriko** [comment]
As the file moved to other location, I would find it clearer to close this MR in favor of new one.

**Pull request closed without merging**